src/util/postgres_client.js

```diff
@@ -268,14 +277,60 @@ class BulkOp {
     }
 
     findAndUpdateOne(find, update) {
-        const pg_update = mongo_to_pg.convertUpdate('data', update);
-        const pg_selector = mongo_to_pg('data', find);
-        const query = `UPDATE ${this.name} SET data = ${pg_update} WHERE ${pg_selector}`;
+        let encoded_update = encode_json(this.schema, update);
+        const pg_update = mongo_to_pg.convertUpdate('data', encoded_update);
+
+        let encoded_find = encode_json(this.schema, find);
+        const pg_selector = mongo_to_pg('data', encoded_find);
+
+        let query;
+        const update_keys = encoded_update.$set && Object.keys(encoded_update.$set).filter(key => key.includes('.$.'));
```

Comment:
This can be queried in other paths as well and needs to be handled just like the types

Reply:
As discussed, I added a comment that specifies .$. is currently handled for this function only as needed in the current code.